Amq_List_Broker-main/template/newcloudformationforlambda.yaml

```diff
@@ -0,0 +1,99 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: "get activemq each topic count"
+Parameters:
+  LambdaVersion:
+    Description: "LambdaFunction Version"
+    Type: String
+    Default: ""
+  brokerUserName:
+    Description: "Enter username to connect mq"
+    Type: String
+    Default: "mq_user_name"
+  brokerPassword:
+    Description: "Enter the password to connect mq broker"
+    Type: String
+    Default: "mq_password"
+  SendToEmail:
+    Description: "Send To Email for Notification"
+    Type: String
+    Default: "njaiswal@cogitocorp.com"
+  clientname:
+    Description: "enter the client name"
+    Type: String
+    Default: ""
+  LambdaBucketName:
+    Description: "Name of S3 Bucket for where Lambda function zip files are uploaded"
+    Type: String
+    Default: "stage-cogito-lambda-bucket"
+    MinLength: 3
+Resources:
+  Lambdarole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Sid: ""
+            Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+                - ses.amazonaws.com
+            Action: sts:AssumeRole
+      RoleName: active-mq-list-lambda-role
+      Policies:
+        - PolicyName: Active-Mq-List-policy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - secretsmanager:*
+                  - ec2:*
+                  - mq:*
+                  - ses:*
+                Resource: "*"
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: "*"
+  LambdaFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: activemq.lambda_handler
+      FunctionName: ActiveMQ_list 
+      Description: "list activemq and send notification"
+      Role: !GetAtt Lambdarole.Arn
+      Code:
+        S3Bucket: !Ref LambdaBucketName
+        S3Key:  Amq_List_Broker.zip
+      VpcConfig:
+         SecurityGroupIds: [sg-003401576ce8c592d,sg-0efb09e99fadd0f89]
```

Comment:
These security groups and subnets will be different for each client which should be passed from outside instead of hardcoded

Reply:
to make user change define the VPC parameter so